### ISSUE TITLE
Bugs/features after the new sidebar

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/server#17737]
After #15719 

## parity
- [x] https://github.com/nextcloud/groupfolders/pull/507
- [x] https://github.com/nextcloud/socialsharing/pull/37
- [x] https://github.com/nextcloud/guests/pull/153
- [ ] ~~Sidebar preview plugins (e.g. from files_pdfviewer) are no longer working~~

## cleanup
- [x] Old templates and scripts from core https://github.com/nextcloud/server/pull/17753
- [x] Migrate scripts loading to the new [LoadSidebar](https://github.com/nextcloud/server/commit/d88b93c919b6bf7db1853ec91a68996e07a75ba4) event https://github.com/nextcloud/server/pull/17751
- [x] Use `OCA.Files.Sidebar.open(path)` instead of file setter https://github.com/nextcloud/server/pull/17750
- [x] Fix files_sharing app LoadSidebar event https://github.com/nextcloud/server/pull/17821
- [x] Fix files_versions app LoadSidebar event https://github.com/nextcloud/server/pull/17829
- [x] Fix comments app init and LoadSidebar event https://github.com/nextcloud/server/pull/17803

## bugs
- [x] Cannot open the sidebar from the share icon breadcrumbs
- [x] The tags should not me shown if none are set, we should add that to the actions menu on the sidebar
- [ ] https://github.com/nextcloud/nextcloud-vue/issues/704
- [x] The currently selected file is not highlighted in the file list anymore

**Comment by juliushaertl** [comment]
Added: 
- The currently selected file is not highlighted in the file list anymore
- Sidebar preview plugins (e.g. from files_pdfviewer) are no longer working

**Comment by skjnldsv** [comment]
> * Sidebar preview plugins (e.g. from files_pdfviewer) are no longer working

this is done on purpose.
they should comply to proper preview generator :)

**Comment by juliushaertl** [comment]
I think that is a regression that people will complain about a lot. I also fear that there is no real way to have a proper [preview of markdown documents generated as an image](https://github.com/nextcloud/text/issues/23) with just PHP and we should have nice previews for documents of the text app for example.

**Comment by juliushaertl** [comment]
Also plaintext previews look really bad imo:
![image](https://user-images.githubusercontent.com/3404133/70240550-b42ddc80-176d-11ea-9a3f-df3f0d348fb3.png)

**Comment by skjnldsv** [comment]
Pdf can be rendered with imagick.
This is the same as videos. We could inject lots of scripts into the sidebar to fake a real preview, but it's:
1. heavy to load the script
2. heavy to load the file
3. complexify the whole thing a lot and is much sloppier
4. can lead to security issues if we load a file that contains a security hack or something (pdf xss...)

**Comment by szaimen** [comment]
> Pdf can be rendered with imagick. This is the same as videos. 

Aren't there problems with imagick?
https://github.com/nextcloud/server/issues/13099 

Apart from that, I aggree with @skjnldsv.

**Comment by wiswedel** [comment]
> Cannot open the sidebar from the share icon breadcrumbs

This is checked as resolved, but still happens in beta2 - has the fix been merged just too recently or is something still wrong?

**Comment by skjnldsv** [comment]
> This is checked as resolved, but still happens in beta2 - has the fix been merged just too recently or is something still wrong?

fix was merged but another pr broke it.

**Comment by jancborchardt** [comment]
What about the **The tags should not me shown if none are set**, it seems to be blocked by "differentiate focus and focus-keyboard on actions" https://github.com/nextcloud/nextcloud-vue/issues/704

Now that @jospoortvliet is taking screenshots, the "Collaborative tags" box is in all the screenshots and it doesn’t look good. cc @skjnldsv :)

**Comment by skjnldsv** [comment]
> What about the **The tags should not me shown if none are set**, it seems to be blocked by "differentiate focus and focus-keyboard on actions" [nextcloud/nextcloud-vue#704](https://github.com/nextcloud/nextcloud-vue/issues/704)

No it's just a matter of adding a toggle for it :(
No time yet

**Comment by juliushaertl** [comment]
> The tags should not me shown if none are set, we should add that to the actions menu on the sidebar

Already fixed by https://github.com/nextcloud/server/pull/18732

**Comment by juliushaertl** [comment]
Closing since all server related ones are fixed and only nextcloud/nextcloud-vue#704 is remaining to be fixed in the components.
